kgforge/core/archetypes/model.py

```diff
@@ -110,6 +118,35 @@ def _template(self, type: str, only_required: bool) -> Dict:
 
     # Mappings.
 
+    def sparql(
+            self, query: str,
+            debug: bool,
+            limit: int = DEFAULT_LIMIT,
+            offset: int = DEFAULT_OFFSET,
+            **params
+    ) -> List[Resource]:
+        rewrite = params.get("rewrite", True)
+
+        qr = handle_sparql_query(
+            query=query,
+            model_context=self.context(),
+            metadata_context=None,  # TODO something else?
+            rewrite=rewrite,
+            limit=limit,
+            offset=offset,
+            default_limit=DEFAULT_LIMIT,
+            default_offset=DEFAULT_OFFSET,
+            debug=debug
+        )
+
+        return self._sparql(qr)
+
+    def _sparql(self, query: str) -> SPARQLResult:
```

Comment:
This is the unprocessed return value and type from the graph object being queried, I'm not sure which format is the most appropriate